hapi-fhir-storage/src/main/java/ca/uhn/fhir/jpa/api/svc/IDeleteExpungeSvc.java

```diff
@@ -19,13 +19,15 @@
  */
 package ca.uhn.fhir.jpa.api.svc;
 
+import ca.uhn.fhir.model.api.IModelJson;
 import ca.uhn.fhir.rest.api.server.storage.IResourcePersistentId;
 
 import java.util.List;
 
 public interface IDeleteExpungeSvc<T extends IResourcePersistentId<?>> {
 
-	int deleteExpunge(List<T> thePersistentIds, boolean theCascade, Integer theCascadeMaxRounds);
+	<PT extends IModelJson> int deleteExpunge(
+			List<T> thePersistentIds, boolean theCascade, Integer theCascadeMaxRounds, PT theWorkChunk);
```

Comment:
major problem:  This is a breaking change with Mongo.  You'll need a Smile change too.

suggestion: Just pass the String chunkId rather than the whole chunk.  That allows you to skip all the generic noise.

If you want to avoid a version bump, I can show you how to do that with a default method.  Something like this.
```suggestion
default int deleteExpunge(
			List<T> thePersistentIds, boolean theCascade, Integer theCascadeMaxRounds, String theWorkChunkId) {
			// delete body once Mongo is fixed.
			return deleteExpunge(thePersistentIds, theCascade, theCascadeMaxRounds);
			}
@Deprecated // to delete asap
default int deleteExpunge(
			List<T> thePersistentIds, boolean theCascade, Integer theCascadeMaxRounds) {
			return deleteExpunge(thePersistentIds, theCascade, theCascadeMaxRounds, "unknown");
			}
```

Reply:
Forgot to post the Smile change... that is my bad.